File: bestia-models/src/main/java/net/bestia/model/dao/MapParameterDAO.java
package net.bestia.model.dao;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import net.bestia.model.domain.MapParameter;

/**
 * Simple DAO object for accessing the map parameter data.
 * 
 * @author Thomas Felix
 *
 */
@Repository("mapParameterDao")
public interface MapParameterDAO extends CrudRepository<MapParameter, Integer> {

	/**
	 * Returns the latest {@link MapParameter} from the database.
	 * 
	 * @return The latest {@link MapParameter}.
	 */
	@Query("FROM MapParameter mp ORDER BY mp.id DESC")
	MapParameter findLatest();
}

File: bestia-models/src/main/java/net/bestia/model/dao/MapParameterDAO.kt
package net.bestia.model.dao

import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

import net.bestia.model.domain.MapParameter

/**
 * Simple DAO object for accessing the map parameter data.
 *
 * @author Thomas Felix
 */
@Repository("mapParameterDao")
interface MapParameterDAO : CrudRepository<MapParameter, Int> {

  /**
   * Returns the latest [MapParameter] from the database.
   *
   * @return The latest [MapParameter].
   */
  fun findFirstByOrderByIdDesc(): MapParameter?
}
